### ISSUE TITLE
Flatpak install issue

### ISSUE THREAD

**Issue by 00willo** [projecthamster/hamster#687]
I'm very new to using flatpak, but I thought I'd give the flatpak build a go.

Followed the README, building from master.

```
flatpak-builder --force-clean --user --install build/flatpak org.gnome.Hamster.json
```
The [build output](https://gist.github.com/00willo/a39781026a08b50251d0100b47ea7db4) looked good and it appeared to install successfully.

```
$flatpak list
Name                                                Application ID                               Version        Branch        Origin                Installation
default                                             org.freedesktop.Platform.GL.default                         19.08         flathub               user
Intel                                               org.freedesktop.Platform.VAAPI.Intel                        19.08         flathub               user
Hamster                                             org.gnome.Hamster                                           master        hamster-origin        user
GNOME Application Platform version 3.36             org.gnome.Platform                                          3.36          flathub               user
GNOME Software Development Kit version 3.36         org.gnome.Sdk                                               3.36          flathub               user
```
However, I get this error when running it.
```
$ flatpak run org.gnome.Hamster
hamster.client - WARNING - Server and client version mismatch:
    server: 3.0.2
    client: v3.0.2-28-g19d16009

    This is sometimes used during bisections,
    but generally calls for trouble.

    Remember to kill hamster daemons after any version change
    (this is safe):
    pkill -f hamster-service
    pkill -f hamster-windows-service
    see also:
    https://github.com/projecthamster/hamster#kill-hamster-daemons

Traceback (most recent call last):
  File "/app/bin/hamster", line 149, in on_activate_window
    self._open_window(action.get_name(), data)
  File "/app/bin/hamster", line 184, in _open_window
    self.overview_controller = Overview()
  File "/app/lib/python3.7/site-packages/hamster/overview.py", line 477, in __init__
    self.find_facts()
  File "/app/lib/python3.7/site-packages/hamster/overview.py", line 531, in find_facts
    self.facts = self.storage.get_facts(start, end, search_terms=search)
  File "/app/lib/python3.7/site-packages/hamster/client.py", line 156, in get_facts
    for fact in self.conn.GetFactsJSON(dbus_range, search_terms)]
  File "/app/lib/python3.7/site-packages/dbus/proxies.py", line 147, in __call__
    **keywords)
  File "/app/lib/python3.7/site-packages/dbus/connection.py", line 653, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Python.sqlite3.OperationalError: Traceback (most recent call last):
  File "/usr/lib64/python3.9/site-packages/dbus/service.py", line 715, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/libexec/hamster/hamster-service", line 335, in GetFactsJSON
  File "/usr/lib/python3.9/site-packages/hamster/storage/storage.py", line 161, in get_facts
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 732, in __get_facts
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 900, in fetchall
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 887, in get_connection
sqlite3.OperationalError: unable to open database file

hamster.client - WARNING - Server and client version mismatch:
    server: 3.0.2
    client: v3.0.2-28-g19d16009

    This is sometimes used during bisections,
    but generally calls for trouble.

    Remember to kill hamster daemons after any version change
    (this is safe):
    pkill -f hamster-service
    pkill -f hamster-windows-service
    see also:
    https://github.com/projecthamster/hamster#kill-hamster-daemons

Traceback (most recent call last):
  File "/app/bin/hamster", line 149, in on_activate_window
    self._open_window(action.get_name(), data)
  File "/app/bin/hamster", line 184, in _open_window
    self.overview_controller = Overview()
  File "/app/lib/python3.7/site-packages/hamster/overview.py", line 477, in __init__
    self.find_facts()
  File "/app/lib/python3.7/site-packages/hamster/overview.py", line 531, in find_facts
    self.facts = self.storage.get_facts(start, end, search_terms=search)
  File "/app/lib/python3.7/site-packages/hamster/client.py", line 156, in get_facts
    for fact in self.conn.GetFactsJSON(dbus_range, search_terms)]
  File "/app/lib/python3.7/site-packages/dbus/proxies.py", line 147, in __call__
    **keywords)
  File "/app/lib/python3.7/site-packages/dbus/connection.py", line 653, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: org.freedesktop.DBus.Python.sqlite3.OperationalError: Traceback (most recent call last):
  File "/usr/lib64/python3.9/site-packages/dbus/service.py", line 715, in _message_cb
    retval = candidate_method(self, *args, **keywords)
  File "/usr/libexec/hamster/hamster-service", line 335, in GetFactsJSON
  File "/usr/lib/python3.9/site-packages/hamster/storage/storage.py", line 161, in get_facts
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 732, in __get_facts
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 900, in fetchall
  File "/usr/lib/python3.9/site-packages/hamster/storage/db.py", line 887, in get_connection
sqlite3.OperationalError: unable to open database file
```
I think, I've worked out that the db file should be here - `~/.var/app/org.gnome.Hamster/data/` and it's empty.

Is there a bug here? Is something missing from the `README` that I need to do to initialise the sqlite file?
I tried copying the `hamster.db` into the path but it still failed with the same issue.

**Comment by matthijskooijman** [comment]
It's been a while, but did you ever resolve this? I've just tried the flatpak build myself, and it created a new sqlite database as expected. I do note that your debug output indicates a missing version, so I wonder if you maybe have a locally-installed version in addition to the flatpak build, and maybe those are conflicting somehow?

**Comment by matthijskooijman** [comment]
@00willo, could you reply to my previous message? Otherwise I think we should assume this is an issue in your local environment and close this issue.

**Comment by DirkHoffmann** [comment]
@00willo, generation of a flatpak file is now included by default in new releases, see [3.0.3](https://github.com/projecthamster/hamster/releases/tag/v3.0.3) "Assets" at the very bottom.
Hence I also believe that we can close this issue. Please comment, if you disagree.
